core/primitives/decode_stream.hpp

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef FILECOIN_CORE_PRIMITIVES_DECODE_STREAM_HPP
+#define FILECOIN_CORE_PRIMITIVES_DECODE_STREAM_HPP
+
+#include <string_view>
+
+#include "common/outcome.hpp"
+#include "common/hexutil.hpp"
+#include "common/textutil.hpp"
+
+namespace primitives {
+
+  /**
+   * Base class for representing data streams
+   */
+  template <typename T>
+  class DecodeStream {
+   public:
+    static constexpr auto is_decoder_stream = true; /**< Decoder type flag */
+
+    /**
+     * @brief Decode content from hex string
+     * @param data input string
+     * @return decoded data
+     */
+    DecodeStream<T> &operator>>(std::string_view data) {
+      fc::outcome::result<std::vector<uint8_t>> bytes = fc::common::unhex(data);
+      if (bytes) {
+        bytes >> *this;
```

Comment:
I don't know why it compiles. Outcome result is not value itself, you have to take bytes.value() and it will be rvalue-reference to std::vector<uint8_t>. But before that you need to check whether it is value or error. If it is error, you cannot go on with it normally, in case of streams you'd better throw an exception, provided that somebody will catch it later. Pls take a look at Kagome's core/common/outcome_throw.hpp: raise() and its usages.

Reply:
This method was unused and now refactored - overloaded operators can't throw exceptions, all error handling executing in the free wrapper functions in the leb128.hpp